playwright/src/test/java/com/microsoft/playwright/TestSoftLocatorAssertions.java

```diff
@@ -0,0 +1,252 @@
+package com.microsoft.playwright;
+
+import com.microsoft.playwright.assertions.LocatorAssertions;
+import com.microsoft.playwright.impl.LocatorAssertionsImpl;
+import com.microsoft.playwright.impl.LocatorAssertionsImplProxy;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static com.microsoft.playwright.Utils.createProxy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.verify;
+
+// The only thing we want to verify in these tests is that the correct method was called
+@ExtendWith(MockitoExtension.class)
+public class TestSoftLocatorAssertions {
+  @Mock
+  private LocatorAssertionsImpl locatorAssertionsMock;
+  private final static Pattern pattern = Pattern.compile("");
+  private final static Pattern[] patternArray = new Pattern[]{};
+  private final static String[] stringArray = new String[]{};
+  private LocatorAssertionsImplProxy proxy;
+
+  @BeforeEach
+  void beforeEach() {
+    proxy = createProxy(LocatorAssertionsImplProxy.class, locatorAssertionsMock);
+  }
+
+  @Test
+  void proxyImplementsLocatorAssertions() {
+    assertTrue(LocatorAssertions.class.isAssignableFrom(proxy.getClass()));
+  }
+
+  @Test
+  void not() {
+    proxy.not();
+    verify(locatorAssertionsMock).not();
+  }
+
+  @Test
+  void isAttached() {
+    assertDoesNotThrow(() -> proxy.isAttached(null));
+    verify(locatorAssertionsMock).isAttached(null);
+  }
+
+  @Test
+  void isChecked() {
+    assertDoesNotThrow(() -> proxy.isChecked(null));
+    verify(locatorAssertionsMock).isChecked(null);
+  }
+
+  @Test
+  void isDisabled() {
+    assertDoesNotThrow(() -> proxy.isDisabled(null));
+    verify(locatorAssertionsMock).isDisabled(null);
+  }
+
+  @Test
+  void isEditable() {
+    assertDoesNotThrow(() -> proxy.isEditable(null));
+    verify(locatorAssertionsMock).isEditable(null);
+  }
+
+  @Test
+  void isEmpty() {
+    assertDoesNotThrow(() -> proxy.isEmpty(null));
+    verify(locatorAssertionsMock).isEmpty(null);
+  }
+
+  @Test
+  void isEnabled() {
+    assertDoesNotThrow(() -> proxy.isEnabled(null));
+    verify(locatorAssertionsMock).isEnabled(null);
+  }
+
+  @Test
+  void isFocused() {
+    assertDoesNotThrow(() -> proxy.isFocused(null));
```

Comment:
i wrote these tests because i had fat fingered a couple of the calls.  i think they do provide value when new assertions are added and leaves less up to chance.  I moved away from a lot of direct tests of the assertions as I do in `TestSoftAssertions` because i found myself duplicating the same error messages as the actual assertion tests.  If those messages ever change, we would need to update in more places.